core/common/logger.go

```diff
@@ -0,0 +1,69 @@
+package common
+
+import (
+	"fmt"
+	"log/slog"
+	"os"
+
+	"github.com/cometbft/cometbft/libs/log"
+)
+
+type Logger struct {
+	log slog.Logger
+}
+
+func NewLogger(opts *slog.HandlerOptions) *Logger {
+	logger := *slog.New(slog.NewJSONHandler(os.Stdout, opts))
+	return &Logger{
+		logger,
+	}
+}
+
+func (l *Logger) Debug(msg string, keyvals ...interface{}) {
+	l.log.Debug(msg, keyvals...)
+}
+
+func (l *Logger) Info(msg string, keyvals ...interface{}) {
+	l.log.Info(msg, keyvals...)
+}
+
+func (l *Logger) Error(msg string, keyvals ...interface{}) {
+	l.log.Error(msg, keyvals...)
+}
+
+func (l *Logger) With(keyvals ...interface{}) log.Logger {
+	newLogger := l.log.With(keyvals...)
+	return &Logger{log: *newLogger}
+}
+
+func (l *Logger) Debugf(msg string, keyvals ...interface{}) {
+	message := fmt.Sprintf(msg, keyvals...)
+	l.log.Debug(message)
+}
+
+func (l *Logger) Errorf(msg string, keyvals ...interface{}) {
```

Comment:
buncha methods on a custom logger instance so we can pass this logger to any library and get the same json formatting